### PR TITLE
Offer details

### PR DESCRIPTION
**Pull request by lightme-fan**
# Description
Showing the offer details when the user clicks on the see-details button on an offer.

## Type of changes
- [ ] Add a simple cypress test file called `OfferDetails.spec.js`.
- [ ] Add a Route in the `App.tsx` to redirect the user to the Offer details.
- [ ] Add OFFER path in the `Paths` folder.
- [ ] Add `id` and `offerId` type which are strngs.
- [ ] Add `OfferDetails` component with its styles by using styled-component.
- [ ] Add a delete and edit icon.
- [ ] Add `OfferDetailPage`
- [ ] Add a new redux slice to get a specific offer.
- [ ] Fetch an offer by Id. 

## Screenshots
- [ ] Mobile vew
Screenshot 2022-02-28 09:38:45

- [ ] Desktop view
Screenshot 2022-02-28 09:39:43

# Checklist
- [x] I navigate to (/offer/:id) by clicking on one of the offer list items or pasting url directly into the browser.
- [x] Offer details page is displayed
- [x] I can see offer details and photo gallery.
- [x] I can see details of the seller.
- [x] I'm able to send an email to the user with prefilled.
- [x] I'm able to see the contact phone number after clicking 'reveal phone number'.
- [x] I'm able to see all images for the offer with their description.

### CODE REVIEW

**Review by mitia-anah — APPROVED** [review]
Every thing looks good for me

File: src/redux/slices/adsSlice.tsx [hunk 1]
```diff
@@ -85,11 +85,11 @@ export const adsSlice = createSlice({
 export const selectAds = (state: RootState) => state.ads.ads
 export const selectStatus = (state: RootState) => state.ads.status
 
-export const adsSelector = createSelector<RootState, any[], any[]>(
+export const adsSelector = createSelector<any[], any[]>(
   selectAds,
   (ads) => ads
 )
-export const adsStatusSelector = createSelector<RootState, string, string>(
+export const adsStatusSelector = createSelector<any[], string>(
```
Comment (andrzej-onja) [inline src/redux/slices/adsSlice.tsx:92]: have you tried to remove the `<any[], string>` all together ? or type it correctly with no any

File: src/redux/slices/offerDetailsSlice.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    }
+}
+export const offerDetailsSlice = createSlice({
+    name: 'offer',
+    initialState,
+    reducers: {},
+    extraReducers: (builder) => {
+        builder.addCase(fetchOfferDetails.fulfilled, (state, { payload }) => {
+            state.offerDetails = payload.getad
+        })
```
Comment (andrzej-onja) [inline src/redux/slices/offerDetailsSlice.tsx:80]: we should have a case for loading and error

Comment (andrzej-onja) [reply src/redux/slices/offerDetailsSlice.tsx:80]: use loading to display indicator, use error to display simple message

File: src/redux/slices/userSlice.tsx [hunk 3]
```diff
@@ -140,21 +140,21 @@ const selectRegisterError = (state: RootState) => state?.user?.registerError
...
-export const userErrorRegister = createSelector<RootState, any, any>(
+export const userErrorRegister = createSelector<any[], any>(
   selectRegisterError,
   (registerError) => registerError
 )
-export const userStatusSelector = createSelector<RootState, any, any>(
+export const userStatusSelector = createSelector<any[], any>(
   selectUserStatus,
   (status) => status
 )
```
Comment (andrzej-onja) [inline src/redux/slices/userSlice.tsx:160]: same here please do not use any

**Review by andrzej-onja — APPROVED** [review]

**Review by bakozetra — APPROVED** [review]

**Pull request merged**